File: ExamFrontend/src/app/services/question/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from '../helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  public evalQuiz(questions: any,username:any) {
    return this.http.post(`${baseUrl}/question/eval-quiz/${username}`,questions)
  }

  constructor(private http:HttpClient) { }

  public viewQuestionsbyQidAdmin(questionId:any){
   return this.http.get(`${baseUrl}/question/quiz/admin/${questionId}`,questionId)
  }

  public viewQuestionsbyQidForTest(qId:any){
    return this.http.get(`${baseUrl}/question/quiz/${qId}`,qId)
   }

  public getQuestion(questionId:any){
    return this.http.get(`${baseUrl}/question/${questionId}`,questionId)
   }

  public deleteQuestion(questionId:any){
    return this.http.delete(`${baseUrl}/question/${questionId}`,questionId)
   }

   public addQuestion(question:any){
    console.log(question)
    return this.http.post(`${baseUrl}/question/`,question);
   }

   public updateQuestion(question:any){
    return this.http.put(`${baseUrl}/question/`,question)
   }
 
 

  

}

File: ExamServer/src/main/java/com/ateaf/ExamServer/service/QuestionService.java
package com.ateaf.ExamServer.service;

import com.ateaf.ExamServer.model.exam.Question;
import com.ateaf.ExamServer.model.exam.Quiz;

import java.util.List;
import java.util.Map;
import java.util.Set;


public interface QuestionService {

    public Question addQuestion(Question question);
    public Question updateQuestion(Question question);
    public Set<Question> getQuestions();
    public Question getQuestion(Long questionId);
    public void deleteQuestion(Long questionId);
    public Set<Question> getQuestionsOfQuiz(Quiz quiz);
    public Set<Question> getQuestionsOfQuizAdmin(Quiz quiz);
    
    public Map<String,Object> evalQuiz(String username, List<Question> questionList);
}
